### DISCUSSION TITLE
android适配问题 4.3.51 PAGImageView加载报错PAGDecoder: Failed  to open SequenceFile

### DISCUSSION THREAD

**Discussion by zengke328** [Tencent/libpag#2242] [Q&A] [answered]
【版本】4.3.51
【官方demo】
【手机型号】三星 Galaxy Z Flip5 SM-F7310
【手机系统】android13 One U版本5.1.1 
【报错信息】PAGDecoder: Failed to open SequenceFile!
【是否必现】必现
【描述】PAGImageView加载本地的assets文件会报错，无法加载。

**Comment by zengke328** [comment]
![3EFF71E5-6548-4be6-80A4-1AEBAC37F007](https://github.com/Tencent/libpag/assets/22447723/41c00bac-9ff9-4a8c-b279-02abcc355176)
![996D0958-32F0-432c-8513-CA60EDABC7F9](https://github.com/Tencent/libpag/assets/22447723/5aeddc2b-dc21-4774-9b82-f41b054ce6dc)

**Comment by kevingpqi123** [comment]
你先看下通过 PAGFile 的方法加载获取到的 PAGFile 是否为空，如果为空，就是加载方法的问题
后续提问题如果是必现的，建议附上复现 demo，这样一次性就可以定位问题，大家效率都高

**Reply by zengke328** [reply]
看了下 PAGFile pagFile = PAGFile.Load(getAssets(), "1.pag");获取到的PAGFile不为空。
我用的官方测试demo： https://github.com/libpag/pag-android

![d73c58f77163dfd57f6be9c6f4acc255](https://github.com/Tencent/libpag/assets/22447723/21ff6d17-9723-4141-b9c8-c584637027cb)

**Reply by zengke328** [reply]
补充下：PAGView可以正常播放assets下pag文件。

**Reply by Yetland** [reply]
我在三星s23测试也有这个问题，PAGView正常，但是PAGImageView就是播放不了。虽然PAGImageView不放不了，但是回调竟然都是正常的。

**Reply by b376787348** [reply]
谢谢

**Reply by kevingpqi123** [reply, marked as answer]
针对三星部分手机 PAGImageView 渲染空白的问题，我们分析了下原因，结论如下：
（1）该问题和缓存路径没有太大关系；
（2）出现问题的原因是三星部分机型 HardwareBuffer 相关实现可能存在一些问题，对于 PAGImageView 而言，pag 在播放动画的过程中，创建了一个 Bitmap，该 Bitmap 通过 Bitmap.wrapHardwareBuffer 的方式创建，Bitmap 占用的内存来源于创建的 HardwareBuffer，播放的过程中会不断将渲染的数据 copy 至创建的HardwareBuffer，然后将 Bitmap 渲染出来，但是在三星的部分机型上 Bitmap 的宽高可以正常读取，却无法正常显示，目前已拉群将该问题反馈至三星

对于 pag 而言，如果这里做处理就要加机型判断，我们倾向于等待三星的问题修复，问题修复后会第一时间知会大家
